k256/src/arithmetic/projective.rs

```diff
@@ -224,6 +232,40 @@ impl ProjectivePoint {
     }
 }
 
+impl Group for ProjectivePoint {
+    type Scalar = Scalar;
+
+    fn random(rng: impl RngCore) -> Self {
+        Self::generator() * &Scalar::generate_vartime(rng)
+    }
+
+    fn identity() -> Self {
+        ProjectivePoint::identity()
+    }
+
+    fn generator() -> Self {
+        ProjectivePoint::generator()
+    }
+
+    fn is_identity(&self) -> Choice {
+        self.ct_eq(&Self::identity())
+    }
+
+    #[must_use]
+    fn double(&self) -> Self {
+        ProjectivePoint::double(self)
+    }
+}
+
+impl group::Curve for ProjectivePoint {
+    type AffineRepr = AffinePoint;
+
+    // TODO(tarcieri): avoid panicking on identity
+    fn to_affine(&self) -> AffinePoint {
+        ProjectivePoint::to_affine(self).unwrap()
+    }
```

Comment:
Presently there's no way of representing the identity as an `AffinePoint`.

Perhaps we should just turn it into an enum with an `AffinePoint::Identity` variant?

Reply:
Added `AffinePoint` identity support in #165